### PR TITLE
Add a changelog entry for 0.3.0.0 release

### PR DESCRIPTION
**Pull request by xsebek**
- bump Cabal Swarm version to `0.3.0.0`
- part of #1012

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,86 @@
...
+#### Debugging
+* Better CESK machine pretty-printing by @byorgey in [#948](https://github.com/swarm-game/swarm/pull/948)
+* Add the full expected location for the data directory by @TristanCacqueray in [#908](https://github.com/swarm-game/swarm/pull/908)
+* Extend logs for warnings by @xsebek in [#982](https://github.com/swarm-game/swarm/pull/982)
+
+#### ???
+* show devices that enable commands by @kostmo in [#899](https://github.com/swarm-game/swarm/pull/899)
+* swarm logo animation by @kostmo in [#1018](https://github.com/swarm-game/swarm/pull/1018)
+* support 'orientationMap' by @kostmo in [#1023](https://github.com/swarm-game/swarm/pull/1023)
+* support custom attributes defined in scenarios by @kostmo in [#1058](https://github.com/swarm-game/swarm/pull/1058)
```
Comment (xsebek) [inline CHANGELOG.md:81]: @kostmo I did not yet figure out where to sort these, suggestions? :mage:

Comment (kostmo) [reply CHANGELOG.md:81]: How about a "Scenario authoring" section that includes:
* #927
* #1023
* #1058

a "UI/CLI enhancements" section that includes:
* #899
* #819
* #851?

and put #1018 in the "New scenarios" section.

Comment (xsebek) [reply CHANGELOG.md]: @kostmo thanks, done! :heavy_check_mark:

**Review by kostmo — APPROVED** [review]
LGTM!

**Comment by xsebek** [comment]
@byorgey I plan to merge this tomorrow, do you have any suggestions? 🙂 

I will tag a release right afterwards.

**Comment by byorgey** [comment]
@xsebek I hope to take a look tonight.  I also have a couple tutorial improvements I would like to sneak into this release --- one of my students found some more issues today.  But there is an ice storm coming through tonight and everything is closed tomorrow so I should have plenty of time to work on it tonight or early tomorrow. :smile:

**Review by byorgey — COMMENTED** [review]
Looks good to me!  Now that we merged #1001 we should also maybe mention the upgrades.

**Review by byorgey — APPROVED** [review]

**Comment by xsebek** [comment]
>  I also have a couple tutorial improvements I would like to sneak into this release --- one of my students found some more issues today.

@byorgey I would leave that to minor releases. :wink: 

We should make those more often, the list of new features in this release is insane. :smile:

**Pull request merged**